### PR TITLE
fix(stepfunctions-tasks): state machine role is missing sagemaker:AddTags permission for SageMakerCreateTransformJob task

### LINKED ISSUE #26012: aws-stepfunctions-tasks: state machine role is missing sagemaker:AddTags permission for SageMakerCreateTransformJob task
### Describe the bug

Using Python, I am trying to create a step functions state machine that runs an AWS SageMaker batch transform job using the `.sync` version of the API like this:

```python
batch_inference_job = sfn_tasks.SageMakerCreateTransformJob(
    self,
    "BatchInferenceTransformJob",
    integration_pattern=sfn.IntegrationPattern.RUN_JOB,
    transform_job_name=sfn.JsonPath.string_at("$.transform_job_name"),
    model_name="xxx",
    ...
)

state_machine = sfn.StateMachine(
    self,
    "BatchInferencePipeline",
    state_machine_name="xxx"
    definition=batch_inference_job
)
```

Note that I am not providing anything in the `role` parameter when I instantiate the `StateMachine`. When I try to execute the state machine, I get the following type of error:
```
User: arn:aws:sts::xxx:assumed-role/xxx is not authorized to perform: sagemaker:AddTags on resource: arn:aws:sagemaker:eu-west-1:xxx:transform-job/xxx because no identity-based policy allows the sagemaker:AddTags action (Service: AmazonSageMaker; Status Code: 400; Error Code: AccessDeniedException; Request ID: xxx; Proxy: null)
```

When I take a closer look at what tags step functions is trying to set for the transform job (I am not setting any tags for the job myself), I see some AWS managed tags, which presumably the step functions service appends:
```
{
    "Key": "MANAGED_BY_AWS",
    "Value": "STARTED_BY_STEP_FUNCTIONS"
}
```

So from my viewpoint it seems that the role generated by CDK for the state machine should already by default include a policy that allows the sagemaker:AddTags action. When I tried spinning up the batch transform job with `sfn.IntegrationPattern.REQUEST_RESPONSE`, step functions didn't try to set any tags and submitting the job worked as expected.

### Expected Behavior

The default role generated by cdk for the step functions state machine should have all the necessary permissions to start a job when using `integration_pattern=sfn.IntegrationPattern.RUN_JOB`, including `sagemaker:AddTags`.

### Current Behavior

Got an error when step functions tried to create the batch transform job:

```
User: arn:aws:sts::xxx:assumed-role/xxx is not authorized to perform: sagemaker:AddTags on resource: arn:aws:sagemaker:eu-west-1:xxx:transform-job/xxx because no identity-based policy allows the sagemaker:AddTags action (Service: AmazonSageMaker; Status Code: 400; Error Code: AccessDeniedException; Request ID: xxx; Proxy: null)
```

### Reproduction Steps

```python
batch_inference_job = sfn_tasks.SageMakerCreateTransformJob(
    self,
    "BatchInferenceTransformJob",
    integration_pattern=sfn.IntegrationPattern.RUN_JOB,
    transform_job_name=sfn.JsonPath.string_at("$.transform_job_name"),
    model_name="xxx",
    ...
)

state_machine = sfn.StateMachine(
    self,
    "BatchInferencePipeline",
    state_machine_name="xxx"
    definition=batch_inference_job
)
```

### Possible Solution

Not tested, but it seems that the policies for the role are added in: https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/sagemaker/create-transform-job.ts#L273. So simply adding a policy statement that allows `sagemaker:AddTags` there.

### Additional Information/Context

_No response_

### CDK CLI Version

2.81.0

### Framework Version

_No response_

### Node.js Version

v18.16.0

### OS

MacOS 12.5

### Language

Python

### Language Version

3.10.9

### Other information

_No response_

### PR DESCRIPTION
**Pull request by tam0ri**
If we specified RUN_JOB as IntegrationPattern prop for SageMakerCreateTransformJob construct, StepFunctions executes SageMaker batch transform job [synchronously](https://docs.aws.amazon.com/step-functions/latest/dg/connect-to-resource.html#connect-sync). In this case, StepFunctions add a tag (key: MANAGED_BY_AWS, value: STARTED_BY_STEP_FUNCTIONS) to the job, so state machine role needs the permission to do that. However, currently CDK does not add the permission automatically.

This PR solves the issue by adding `sagemaker:AddTags` permission to state machine role when RUN_JOB is specified as IntegrationPattern prop.

Closes #26012

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by tam0ri** [comment]
Exemption Request: There is no change against existing snapshots. Also, this is a small fix, so I didn't add a new integ test at this point. Please let me know if it's needed.

**Comment by sumupitchayan** [comment]
> Exemption Request: There is no change against existing snapshots. Also, this is a small fix, so I didn't add a new integ test at this point. Please let me know if it's needed.

@tam0ri I believe an integ test here would be useful.

File: packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/sagemaker/create-transform-job.ts [hunk 1]
```diff
@@ -283,6 +283,18 @@ export class SageMakerCreateTransformJob extends sfn.TaskStateBase {
           ],
         }),
       );
+      policyStatements.push(
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/sagemaker/create-transform-job.ts:286]:
```suggestion
```
A single `push` call can be used for both statements.

File: packages/aws-cdk-lib/aws-stepfunctions-tasks/test/sagemaker/create-transform-job.test.ts [hunk 2]
```diff
@@ -306,3 +308,215 @@ test('create transform job with instance type supplied as JsonPath', () => {
     },
   });
 });
+
+test('required permmisions are granted to service role if RUN_JOB is supplied as service integration pattern', () => {
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-stepfunctions-tasks/test/sagemaker/create-transform-job.test.ts:312]:
```suggestion
test('required permissions are granted to service role if RUN_JOB is supplied as service integration pattern', () => {
```

File: packages/aws-cdk-lib/aws-stepfunctions-tasks/test/sagemaker/create-transform-job.test.ts [hunk 3]
```diff
@@ -306,3 +308,215 @@ test('create transform job with instance type supplied as JsonPath', () => {
...
+            ],
+          },
+        },
+      ],
+      Version: '2012-10-17',
+    },
+  });
+});
+
+test('required permmisions are granted to service role if REQUEST_RESPONSE is supplied as service integration pattern', () => {
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/aws-stepfunctions-tasks/test/sagemaker/create-transform-job.test.ts:444]:
```suggestion
test('required permissions are granted to service role if REQUEST_RESPONSE is supplied as service integration pattern', () => {
```

File: packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions-tasks/test/sagemaker/integ.create-transform-job.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import { App, Stack, StackProps } from 'aws-cdk-lib';
+import { InstanceType, InstanceClass, InstanceSize } from 'aws-cdk-lib/aws-ec2';
+import { SageMakerCreateTransformJob, S3DataType } from 'aws-cdk-lib/aws-stepfunctions-tasks';
+import { StateMachine, IntegrationPattern, DefinitionBody } from 'aws-cdk-lib/aws-stepfunctions';
+
+class StepFunctionsTaskCreateTransformJobIntegStack extends Stack {
+  constructor(scope: App, id: string, props?: StackProps) {
+    super(scope, id, props);
+    const batch_inference_job = new SageMakerCreateTransformJob(this, 'Batch Inference', {
```
Comment (lpizzinidev) [inline packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions-tasks/test/sagemaker/integ.create-transform-job.ts:9]:
```suggestion
    super(scope, id, props);

    const task = new SageMakerCreateTransformJob(this, 'BatchInferenceTask', {
```

File: packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions-tasks/test/sagemaker/integ.create-transform-job.ts [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      },
+      transformResources: {
+        instanceCount: 1,
+        instanceType: InstanceType.of(InstanceClass.M4, InstanceSize.XLARGE),
+      },
+      integrationPattern: IntegrationPattern.RUN_JOB,
+    });
+
+    new StateMachine(this, 'SimpleStateMachine', {
+      definitionBody: DefinitionBody.fromChainable(batch_inference_job),
```
Comment (lpizzinidev) [inline packages/@aws-cdk-testing/framework-integ/test/aws-stepfunctions-tasks/test/sagemaker/integ.create-transform-job.ts:31]:
```suggestion
      definitionBody: DefinitionBody.fromChainable(task),
```

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks (and sorry for the late review) 👍
Looks good, just some minor adjustments are needed.

**Comment by tam0ri** [comment]
@lpizzinidev Thank you for your review! I pushed changes according to your comment. Could you review when you have some time?

**Review by lpizzinidev — APPROVED** [review]
Looks good to me, thanks 👍

**Review by vinayak-kukreja — APPROVED** [review]
Hey @tam0ri , apologies for the delay. Thank you for contributing to AWS CDK.

**Pull request merged**